app/controllers/tasks_controller.rb

```diff
@@ -61,6 +61,20 @@ def destroy
     end
   end
 
+  def toggle
+    if @task.done
```

Comment:
please make sure that you you check for validation before you update it, thanks!